pkg/plugins/localfiles.go

```diff
@@ -0,0 +1,91 @@
+package plugins
+
+import (
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/grafana/grafana/pkg/util"
+)
+
+var _ fs.FS = (*LocalFS)(nil)
+
+type LocalFS struct {
+	m        map[string]*LocalFile
+	basePath string
+}
+
+func NewLocalFS(m map[string]struct{}, basePath string) LocalFS {
+	pfs := make(map[string]*LocalFile, len(m))
+	for k := range m {
+		pfs[k] = &LocalFile{
+			path: k,
+		}
+	}
+
+	return LocalFS{
+		m:        pfs,
+		basePath: basePath,
+	}
+}
+
+func (f LocalFS) Open(name string) (fs.File, error) {
+	cleanPath, err := util.CleanRelativePath(name)
+	if err != nil {
+		return nil, err
+	}
+
+	if kv, exists := f.m[filepath.Join(f.basePath, cleanPath)]; exists {
+		if kv.f != nil {
+			return kv.f, nil
+		}
+		return os.Open(kv.path)
+	}
+	return nil, ErrFileNotExist
+}
+
+func (f LocalFS) Base() string {
+	return f.basePath
+}
+
+func (f LocalFS) Files() []string {
+	var files []string
+	for p := range f.m {
+		r, err := filepath.Rel(f.basePath, p)
+		if strings.Contains(r, "..") || err != nil {
+			continue
+		}
+		files = append(files, r)
+	}
+
+	return files
+}
+
+var _ fs.File = (*LocalFile)(nil)
```

Comment:
what does this line do? (maybe add a comment here?)

Reply:
It's somewhat of a Golang convention to do interface verification. You'll see it littered throughout the Grafana codebase - a kind of handy indicator right away showing what you may be dealing with when inspecting a struct